src/apps/lwaftr/ipv6_apps.lua

```diff
@@ -71,15 +75,37 @@ end
 
 function Reassembler:cache_fragment(frag)
    local cache = self.fragment_cache
+
+   if self.fragment_count >= self.fragment_cache_max then
+      local n = math.random(self.fragment_key_count)
+      for _, frags in pairs(cache) do
```

Comment:
Isn't this O(N) in the number of fragment keys? .... and only called when that number is high? That's a problem.